File: textbook_programs/Liskov/SzuloGyerek/szulogyerek.cpp
#include <iostream>
#include <string>
class Person
{
protected:
    int ageP;
    std::string nameP;

public:
    Person(std::string name, int age)
    {
        nameP = name;
        ageP = age;
    }

    void setAge(int age)
    {
        ageP = age;
    }

    std::string getName()
    {
        return nameP;
    }
    int getAge()
    {
        return ageP;
    }

    void setName(std::string name)
    {
        nameP = name;
    }
};

class Person2 : public Person
{
private:
    std::string emailP;

public:
    Person2(std::string name, int age, std::string email) : Person(name, age)
    {
        emailP = email;
    }
    std::string getEmail()
    {
        return emailP;
    }
};

int main()
{
    Person *person = new Person2("Pisti", 37, "email");
    td::cout << person->getEmail();
}

File: textbook_programs/Liskov/SzuloGyerek/szulogyerek.java
class Person {
    protected int age;
    protected String name;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public int getAge() {
        return this.age;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

class Person2 extends Person {
    private String email;

    public Person2(String name, int age, String email) {
        super(name, age);
        this.email = email;
    }

    public String getEmail() {
        return this.email;
    }

}

class szuloGyerek {

    public static void main(String args[]) {
        Person person = new Person2("Pisti", 37, "email@email.com");

        System.out.println(person.getEmail());
        // System.out.println(((Person2) person).getEmail());
    }
}